sdk/resource_locator.go

```diff
@@ -146,5 +225,32 @@ func (rl *ResourceLocator) readResourceLocator(reader io.Reader) error {
 		return errors.Join(Error("Error reading ResourceLocator body value"), err)
 	}
 	rl.body = string(body) // TODO - normalize to lowercase?
+	// read the identifier if it exists
+	switch rl.protocol & 0xf0 {
+	case identifierNone, urlProtocolHTTPS:
```

Comment:
```suggestion
	case identifierNone:
```

Reply:
we hit exhaustive lint errors. added some comments and lint exceptions